tests/php/modules/masterbar/test-class-wpcom-admin-menu.php

```diff
@@ -0,0 +1,544 @@
+<?php
+/**
+ * Tests for WPcom_Admin_Menu class.
+ *
+ * @package Jetpack
+ */
+
+use Automattic\Jetpack\Dashboard_Customizations\WPcom_Admin_Menu;
+use Automattic\Jetpack\Status;
+
+require_jetpack_file( 'modules/masterbar/admin-menu/class-admin-menu.php' );
+require_jetpack_file( 'modules/masterbar/admin-menu/class-wpcom-admin-menu.php' );
+require_jetpack_file( 'tests/php/modules/masterbar/data/admin-menu.php' );
+
+/**
+ * Class Test_WPcom_Admin_Menu.
+ *
+ * @coversDefaultClass Automattic\Jetpack\Dashboard_Customizations\WPcom_Admin_Menu
+ */
+class Test_WPcom_Admin_Menu extends WP_UnitTestCase {
+
+	/**
+	 * Menu data fixture.
+	 *
+	 * @var array
+	 */
+	public static $menu_data;
+
+	/**
+	 * Submenu data fixture.
+	 *
+	 * @var array
+	 */
+	public static $submenu_data;
+
+	/**
+	 * Test domain.
+	 *
+	 * @var string
+	 */
+	public static $domain;
+
+	/**
+	 * Whether this testsuite is run on WP.com.
+	 *
+	 * @var bool
+	 */
+	public static $is_wpcom;
+
+	/**
+	 * Admin menu instance.
+	 *
+	 * @var WPcom_Admin_Menu
+	 */
+	public static $admin_menu;
+
+	/**
+	 * Mock user ID.
+	 *
+	 * @var int
+	 */
+	private static $user_id = 0;
+
+	/**
+	 * Create shared fixtures.
+	 *
+	 * @param WP_UnitTest_Factory $factory Fixture factory.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		static::$domain  = ( new Status() )->get_site_suffix();
+		static::$user_id = $factory->user->create( array( 'role' => 'administrator' ) );
+
+		static::$menu_data    = get_menu_fixture();
+		static::$submenu_data = get_submenu_fixture();
+	}
+
+	/**
+	 * Set up data.
+	 */
+	public function setUp() {
+		parent::setUp();
+		global $menu, $submenu;
+
+		// Initialize in setUp so it registers hooks for every test.
+		static::$admin_menu = WPcom_Admin_Menu::get_instance();
+
+		$menu    = static::$menu_data;
+		$submenu = static::$submenu_data;
+
+		wp_set_current_user( static::$user_id );
+	}
+
+	/**
+	 * Test get_instance.
+	 *
+	 * @covers ::get_instance
+	 * @covers ::__construct
+	 */
+	public function test_get_instance() {
+		$instance = WPcom_Admin_Menu::get_instance();
+
+		$this->assertInstanceOf( WPcom_Admin_Menu::class, $instance );
+		$this->assertSame( $instance, static::$admin_menu );
+
+		$this->assertSame( 99999, has_action( 'admin_menu', array( $instance, 'reregister_menu_items' ) ) );
+		$this->assertSame( 10, has_action( 'admin_enqueue_scripts', array( $instance, 'enqueue_scripts' ) ) );
+	}
+
+	/**
+	 * Tests add_browse_sites_link.
+	 *
+	 * @covers ::add_browse_sites_link
```

Comment:
According to PHPUnit docs, this usage is not recommended: https://phpunit.readthedocs.io/en/9.3/annotations.html#covers

Please update all the `@covers` usage that does not follow recommendations

Reply:
I am a little confused here. The docs don't recommend  `@covers ClassName::methodName` but on `@coversDefaultClass` docs they have the following which is how our code is structured too

```
/**
 * @coversDefaultClass \Foo\CoveredClass
 */
final class CoversDefaultClassTest extends TestCase
{
    /**
     * @covers ::publicMethod
     */
    public function testSomething(): void
    {
        $o = new Foo\CoveredClass;
        $o->publicMethod();
    }
}
```

which seems fine to me. Is there something I am missing?